### ISSUE TITLE
[Support]: If using zerotier on the host, how should Docker Compose.yml be configured?

### ISSUE THREAD

**Issue by v808comd** [sinamics/ztnet#153]
### 📝 Describe the Problem You Are Having

I tried to delete the paragraph 'zerotier:'. And, change the 'ztnet:' paragraph

Volumes:
-/var/lib/zerotier one:/var/lib/zerotier one
Network_ Mode: host

'Docker compose up - d' will prompt me that the directory is read-only

### 🔖 Version

0.3.9

### 💻 Operating System

None

**Comment by v808comd** [comment]
Doing so may make the IPv6 address of the planet valid

**Comment by sinamics** [comment]
I tested this myself and it worked. 
i Changed the docker-composer as showed bellow, and you also need to change the zerotier url in admin menu to use localhost instead of zerotier hostname.

![image](https://github.com/sinamics/ztnet/assets/7771916/dda27e73-218f-4fce-b841-53acd20b4653)


docker-compose.yml:
```yml
version: "3.1"
services:
  postgres:
    image: postgres:15.2-alpine
    container_name: postgres
    restart: unless-stopped
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: ztnet
    ports:
      - 5432:5432
    volumes:
      - postgres-data:/var/lib/postgresql/data
#    networks:
#      - app-network

#  zerotier:
#    image: zyclonite/zerotier:1.10.6
#    hostname: zerotier
#    container_name: zerotier
#    restart: unless-stopped
#    volumes:
#      - zerotier:/var/lib/zerotier-one
#    cap_add:
#      - NET_ADMIN
#      - SYS_ADMIN
#    devices:
#      - /dev/net/tun:/dev/net/tun
#    networks:
#      - app-network
#    ports:
#      - "9993:9993/udp"
#    environment:
#      - ZT_OVERRIDE_LOCAL_CONF=true
#      - ZT_ALLOW_MANAGEMENT_FROM=172.31.255.0/29

  ztnet:
    image: sinamics/ztnet
    network_mode: host
    container_name: ztnet
    working_dir: /app
    volumes:
      - /var/lib/zerotier-one:/var/lib/zerotier-one
    restart: unless-stopped
#    ports:
#      - 3000:3000
    environment:
      POSTGRES_HOST: localhost
      POSTGRES_PORT: 5432
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: postgres
      NEXTAUTH_URL: "http://localhost:3000"
      NEXTAUTH_SECRET: "random_secret"
#    networks:
#      - app-network
#    links:
#      - postgres
    depends_on:
      - postgres
#      - zerotier
volumes:
  zerotier:
  postgres-data:

networks:
  app-network:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 172.31.255.0/29
```


If you get permission errors, then make sure docker are able to read the /var/lib/zerotier-one folder

**Comment by v808comd** [comment]
Thank you very much, according to your docker composite. yml
At ubuntu 22, the prompt directory is still read-only.
It can be completed in Centos 7,


![dd15e379-5a96-41d8-9b13-4ec4a3a21207](https://github.com/sinamics/ztnet/assets/65583931/1ea481ca-0075-46e8-a12b-35eee806d4b3)

But the browser prompts:
![image](https://github.com/sinamics/ztnet/assets/65583931/8a533976-d71c-4cfc-8db2-fa98e3b9ed90)

**Comment by v808comd** [comment]
Multiple tests,
The browser still reports an error: Internal Server Error.
Unable to enter ztnet.

**Comment by v808comd** [comment]
[root@localhost opt]# docker logs ztnet
psql: error: server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
Postgres is unavailable - sleeping
psql: error: server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
Postgres is unavailable - sleeping
Applying migrations to the database...
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "localhost:5432"

17 migrations found in prisma/migrations

Applying migration `20230402192938_`
Applying migration `20230714081023_flow`
Applying migration `20230717095341_tags`
Applying migration `20230717150852_nw_member`
Applying migration `20230723084736_mailoptions`
Applying migration `20230724093114_nw_description`
Applying migration `20230728082722_init`
Applying migration `20230806115334_zt_central`
Applying migration `20230808053939_db_rework`
Applying migration `20230809202431_cascade`
Applying migration `20230811063619_user_options`
Applying migration `20230813195808_user_group`
Applying migration `20230815175550_custom_root`
Applying migration `20230818113722_custom_root_extend`
Applying migration `20230823185550_update_cascade_delete`
Applying migration `20230825053528_user_invitation`
Applying migration `20230910161252_authorize_warning`

The following migrations have been applied:

migrations/
  └─ 20230402192938_/
    └─ migration.sql
  └─ 20230714081023_flow/
    └─ migration.sql
  └─ 20230717095341_tags/
    └─ migration.sql
  └─ 20230717150852_nw_member/
    └─ migration.sql
  └─ 20230723084736_mailoptions/
    └─ migration.sql
  └─ 20230724093114_nw_description/
    └─ migration.sql
  └─ 20230728082722_init/
    └─ migration.sql
  └─ 20230806115334_zt_central/
    └─ migration.sql
  └─ 20230808053939_db_rework/
    └─ migration.sql
  └─ 20230809202431_cascade/
    └─ migration.sql
  └─ 20230811063619_user_options/
    └─ migration.sql
  └─ 20230813195808_user_group/
    └─ migration.sql
  └─ 20230815175550_custom_root/
    └─ migration.sql
  └─ 20230818113722_custom_root_extend/
    └─ migration.sql
  └─ 20230823185550_update_cascade_delete/
    └─ migration.sql
  └─ 20230825053528_user_invitation/
    └─ migration.sql
  └─ 20230910161252_authorize_warning/
    └─ migration.sql

All migrations have been successfully applied.
npm notice
npm notice New major version of npm available! 9.6.7 -> 10.1.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.1.0>
npm notice Run `npm install -g npm@10.1.0` to update!
npm notice
Migrations applied successfully!
Seeding the database...
Environment variables loaded from .env
Running seed command `ts-node --compiler-options {"module":"CommonJS"} prisma/seed.ts` ...
Seeding User Options complete!
Seeding User Options complete!

🌱  The seed command has been executed.
Database seeded successfully!
Executing command
- info Loaded env from /app/.env
Listening on port 3000 url: http://localhost.localdomain:3000
[root@localhost opt]#

**Comment by v808comd** [comment]
Executing command
- info Loaded env from /app/.env
Listening on port 3000 url: http://localhost.localdomain:3000
Error: connect ECONNREFUSED ::1:34159
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 34159
}
Error: connect ECONNREFUSED ::1:34159
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 34159
